File: backend/app/schemas/carbon.py
from datetime import datetime

from pydantic import BaseModel


class CarbonIntensityPoint(BaseModel):
    from_time: datetime
    to_time: datetime

    forecast: int
    actual: int | None = None

    index: str


class CarbonCurrentResponse(BaseModel):
    provider: str
    region: str
    unit: str

    intensity: CarbonIntensityPoint


class CarbonForecastResponse(BaseModel):
    provider: str
    region: str
    unit: str

    points: list[CarbonIntensityPoint]

File: frontend/src/types/carbon.ts
export type CarbonIntensityPoint = {
  from_time: string;
  to_time: string;

  forecast: number;
  actual: number | null;

  index: string;
};


export type CarbonCurrentResponse = {
  provider: string;
  region: string;
  unit: string;

  intensity: CarbonIntensityPoint;
};


export type CarbonForecastResponse = {
  provider: string;
  region: string;
  unit: string;

  points: CarbonIntensityPoint[];
};